tests/modules/general/test-All_in_One_SEO_Pack_GetMainDescription.php

```diff
@@ -0,0 +1,253 @@
+			$aioseop_options[ $key_config ] = $value_config;
+		}
+		$aioseop_class = new All_in_One_SEO_Pack();
+
+		$args = array(
+			'post_type'      => 'post',
+			'posts_per_page' => ( $this->post_amount + $this->post_amount_excerpt ),
+		);
+
+		$t_posts = get_posts( $args );
+
+		foreach ( $t_posts as $t_post ) {
+			// Admin - On post Edit Screen.
+			//set_current_screen( 'post.php?post=' . $t_post->ID . '&action=edit' );
+			// Post Screen.
+			$this->go_to( get_permalink( $t_post->ID ) );
+
+			$raw_aioseop_description       = get_post_meta( $t_post->ID, '_aioseop_description' );
+			$post_meta_aioseop_description = $raw_aioseop_description[0];
+
+			$test_description = $aioseop_class->get_main_description( $post );
+
+			$count_expected   = 0;
+			$test_desc_length = strlen( $test_description );
+			$message          = 'Description Count FAILED in ' . basename( __FILE__ ) . ' --- ';
+			if ( empty( $post_meta_aioseop_description ) ) {
+				if ( 'on' === $aioseop_options['aiosp_generate_descriptions'] ) {
+					if ( $aioseop_options['aiosp_dont_truncate_descriptions'] ) {
+						$message .= '"' . $test_description . '" ( $test_description ) should be the same as "' . $t_post->post_content . '" ( Post_Content ).';
+						$this->assertSame( $t_post->post_excerpt, $test_description, $message );
+					} else {
+						// 320 length, but incremented to avoid "Less Than Or Equal To"; simplified operation.
+						$count_expected = 321;
+
+						$message .= '$aioseop_options config: ' . print_r( $aioseop_options_config, true ) . '. ';
+						$message .= 'Expected Desc. Length: ' . $count_expected . '. ';
+						$message .= 'Actual Desc. Length: ' . $test_desc_length . '. ';
+						$this->assertLessThan( $count_expected, $test_desc_length, $message );
+					}
+//						if ( $aioseop_options['aiosp_skip_excerpt'] ) {
+//							$this->assertEmpty( $test_description, $message  );
+//						} else {
+//							$this->assertEmpty( $test_description, $message  );
+//						}
+				} elseif ( empty( $aioseop_options['aiosp_generate_descriptions'] ) || 'off' === $aioseop_options['aiosp_generate_descriptions'] ) {
+					$message .= 'Not Empty. Actual Test Description: "' . $test_description . '"';
+					if ( empty( $post->post_excerpt ) ) {
+						$this->assertEmpty( $test_description, $message );
+					} else {
+						$this->assertSame( $post->post_excerpt, $test_description, $message );
+					}
+				}
+			} else {
+				$message .= '"' . $test_description . '" ( $test_description ) should be the same as "' . $post_meta_aioseop_description . '" ( Post meta \'_aioseop_description\' ).';
+				$this->assertSame( $post_meta_aioseop_description, $test_description, $message );
+			}
+		}
+	}
+
+	/**
+	 * Data Provider for (test_) get_main_description
+	 *
+	 * Provides data for the testing function using PHPUnit's annotation.
+	 *
+	 * @since 2.4.4.1
+	 *
+	 * @return array
+	 */
+	public function dataProvider_get_main_description() {
```

Comment:
@EkoJR is mixing two cases (camel and snake) advisable?

Reply:
I debated myself on this as well. In this case, since it was PHPUnit annotations, I had added it to mirror it to try and as an indicator to make it easier to find.

Reply:
I don't think its a good option. Can you please remove it?